doi/src/main/java/ca/nrc/cadc/doi/VospaceDoiClient.java

```diff
@@ -178,19 +185,26 @@ public Resource getResource(String doiSuffixString, String doiFilename)
     public boolean isCallerAllowed(Node node, Subject adminSubject) {
         boolean isRequesterNode = false;
         if (this.includePublicNodes && node.isPublic != null && node.isPublic) {
-            isRequesterNode = true;
-        } else {
-            X500Principal adminX500 = AuthenticationUtil.getX500Principal(adminSubject);
-            String requester = node.getPropertyValue(DoiAction.DOI_VOS_REQUESTER_PROP);
-            log.debug("requester for node: " + requester);
-            if (callersNumericId != null && StringUtil.hasText(requester)) {
-                isRequesterNode = requester.equals(callersNumericId.toString());
-                Set<X500Principal> xset = AuthenticationUtil.getCurrentSubject().getPrincipals(X500Principal.class);
-                for (X500Principal p : xset) {
-                    isRequesterNode = isRequesterNode || AuthenticationUtil.equals(p, adminX500);
-                }
+            return true;
+        } else if (publisherGroupURI != null) {
+            GMSClient gmsClient = new GMSClient(gmsResourceID);
+
+            if (gmsClient.isMember(publisherGroupURI)) {
+                return true;
+            }
+        }
+
+        X500Principal adminX500 = AuthenticationUtil.getX500Principal(adminSubject);
+        String requester = node.getPropertyValue(DoiAction.DOI_VOS_REQUESTER_PROP);
+        log.debug("requester for node: " + requester);
+        if (callersNumericId != null && StringUtil.hasText(requester)) {
+            isRequesterNode = requester.equals(callersNumericId.toString());
```

Comment:
If the caller is the requester this should return true, no need to go further.
If the requester check doesn't return true, then check if the caller is doiadmin and return true.
If all checks fail, then return false. Don't need isRequesterNode.